browser/resources/settings/email_aliases_page/tests/email_aliases.test.tsx

```diff
@@ -0,0 +1,261 @@
+// Copyright (c) 2025 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+import * as React from 'react'
+import { render, screen, waitFor, act } from '@testing-library/react'
+import { ManagePageConnected } from '../email_aliases'
+import { localeRegex } from './test_utils'
+import {
+  AuthenticationStatus,
+  EmailAliasesServiceInterface,
+  EmailAliasesServiceObserverInterface,
+  Alias
+} from 'gen/brave/components/email_aliases/email_aliases.mojom.m'
+
+jest.mock('$web-common/locale', () => ({
+  getLocale: (key: string) => {
+    return key
+  },
+  formatMessage: (key: string, params: Record<string, string>) => {
+    return key
+  },
+  formatLocale: (key: string, params: Record<string, string>) => {
+    return key
+  }
+}))
+
+// Mock the email aliases service
+class MockEmailAliasesService extends EmailAliasesServiceInterface {
+  private observers: Set<EmailAliasesServiceObserverInterface> = new Set()
+  private aliases: Map<string, Alias> = new Map()
+  private authStatus: AuthenticationStatus = AuthenticationStatus.kStartup
+  private authEmail: string = ''
+
+  private notifyObserversAliasesUpdated = () => {
+    this.observers.forEach((observer) => {
+      observer.onAliasesUpdated(Array.from(this.aliases.values()))
+    })
+  }
+
+  private notifyObserversAuthStateChanged = () => {
+    this.observers.forEach((observer) => {
+      observer.onAuthStateChanged({
+        status: this.authStatus,
+        email: this.authEmail
+      })
+    })
+  }
+  addObserver(observer: EmailAliasesServiceObserverInterface) {
+    this.observers.add(observer)
+    this.notifyObserversAliasesUpdated()
```

Comment:
did I miss this before or is this new? Why are we calling these methods when the observer is added? We want to start with a clean slate and clearly specific any test setup for each test. We should also test that the state is as expected before any observer methods fire